src/diffusers/modular_pipelines/components_manager.py

```diff
@@ -166,6 +166,7 @@ def __call__(self, hooks, model_id, model, execution_device):
         device_module = getattr(torch, device_type, torch.cuda)
         mem_on_device = device_module.mem_get_info(execution_device.index)[0]
         mem_on_device = mem_on_device - self.memory_reserve_margin
+
```

Comment:
ohh, what happended here? 
https://github.com/huggingface/diffusers/pull/12566#discussion_r2482332439
It was a good catch, I think we should also add a warning sooner during `enable_model_offload()`

Reply:
My bad for misunderstanding!

I have added the error catching. I added a TODO in `enable_auto_cpu_offload()` to error out early for `mem_get_info()`.